cms/djangoapps/api/docs/decisions/001-course-apps.rst

```diff
@@ -0,0 +1,222 @@
+introduced as a new type of Open edX plugin. Any functionality that can be
+enabled or disabled at a course-level and can be configured by instructors is
+a candidate for being bundled as a Course App.
+
+To be able to show these Course Apps to course admins, they will need to provide
+some bits of metadata, such as a name, a description etc. Additionally we will
+need a common interface for such assps so they can be enabled/disabled
+uniformly.
+
+To do this we can follow the example of existing plugins, [such as Course
+Tabs](https://github.com/edx/edx-platform/blob/636b2ca4c5add531cfce755fdb8965599acd79e0/common/lib/xmodule/xmodule/tabs.py#L24-L243),
+which provide a specific Python class that the plugin can inherit from, or
+implement. The required metadata and features, can be implemented as class
+attributes, and methods on this class.
+
+We can then discover the installed apps using the existing tooling for plugins
+using a subclas of PluginManager designed for this purpose. Here is an example
+for [Course
+Tabs](https://github.com/edx/edx-platform/blob/636b2ca4c5add531cfce755fdb8965599acd79e0/openedx/core/lib/course_tabs.py#L13-L47)
+
+It might not always make sense for an app installed in this way to be
+automatically show up for use on all courses. So each app will expose a method
+to check if it should be available for a particular course. For all new apps
+this should be done via a CourseWaffleFlag.
+
+Once an app has been marked as available for a course, it will show up in the
+API, where the next step is to mark it as enabled.
+
+In the context of these apps, we need to distinguish between *installed*,
+*available* and *enabled*.
+
+Let's look at an existing feature to explain those terms. The `edxnotes` app
+has [code that is part of the
+platform](https://github.com/edx/edx-platform/tree/636b2ca4c5add531cfce755fdb8965599acd79e0/lms/djangoapps/edxnotes).
+This code comes preinstalled since it's part of the platform. So it is already
+*installed*, however no one can use it just yet, since it it needs to first be
+enabled globally. In the case of an external plugin, you consider it installed
+if it is `pip install`ed in the same environment in which edx-platform is
+running.
+
+To make the feature available for use, you need to now [enable a feature
+flag](https://github.com/edx/edx-platform/blob/636b2ca4c5add531cfce755fdb8965599acd79e0/lms/envs/common.py#L531-L543).
+Until this is set course authors/admins will [not even see the option of
+enabling this for their
+course](https://github.com/edx/edx-platform/blob/636b2ca4c5add531cfce755fdb8965599acd79e0/cms/djangoapps/models/settings/course_metadata.py#L91-L93).
+For course apps this is where the availability check comes in.
+
+Finally, once `edxnotes` has been enabled via the feature flag, an option will
+now show up in the advanced settings page of studio that allows you to enable
+it for a particular course. If this value is true, then edxnotes will be enabled
```

Comment:
Sorry, I am not able to understand this whole paragraph. Suddenly we shifted from explaining what is installed/available to how an edxnotes can be enabled in "advanced settings in the studio". 
This paragraph needs to be revisted.